adhocracy4/filters/filters.py

```diff
@@ -57,40 +58,41 @@ def __init__(self, data, *args, **kwargs):
         for key, value in self.defaults.items():
             if key not in data:
                 data[key] = value
-
         super().__init__(data, *args, **kwargs)
 
 
 class DistinctOrderingFilter(django_filters.OrderingFilter):
     """Makes sure, that every queryset gets a distinct ordering, even if
     field to order by (e.g. comment count) would produce a non-distinct
-    odering. In case of random ordering (?), use the extra argument
-    random_seed in order to allow reproducable random orderings.
-    Note: order reproducability relies on the string representation of the seed
+    ordering.
     """
 
-    def __init__(self, *args, **kwargs):
-        self.random_seed = kwargs.pop('random_seed', 'static_randomness')
-        super().__init__(*args, **kwargs)
+    def filter(self, qs, value):
+        if value in django_filters.constants.EMPTY_VALUES:
+            return qs.order_by('pk')
+
+        ordering = [self.get_ordering_value(param) for param in value] + ['pk']
+        return qs.order_by(*ordering)
+
+
+class DistinctOrderingWithDailyRandomFilter(DistinctOrderingFilter):
+    """Note: order reproducability relies on the string representation of the
+    seed
+    """
 
     def filter(self, qs, value):
-        if value == ['?']:
+        if value == ['dailyrandom']:
```

Comment:
We need to change the value in CE then, right? Maybe we keep the '?' for now to be sure not to break anything else.? (Even though I argued against the '?' when we added it in the first place. :smile: )

Reply:
I did this on purpose here, because the '?' is a default for random ordering. So without the `if value == [?]` and then all the stuff we do with the date, it would just do a random order.. this is also what it does now if another ordering filter uses '?', like before. I thought its better to distinguish here, because it is not just the default random ordering.. do you know what I mean?

Reply:
Yes! Nice!